pkg/osbuild/librepo_source.go

```diff
@@ -0,0 +1,120 @@
+package osbuild
+
+import (
+	"fmt"
+
+	"github.com/osbuild/images/pkg/rpmmd"
+)
+
+// LibrepoSource wraps the org.osbuild.librepo osbuild source
+type LibrepoSource struct {
+	Items   map[string]*librepoSourceItem `json:"items"`
+	Options *librepoSourceOptions         `json:"options"`
```

Comment:
I'd prefer we make the types exported/public so that one can, if they want, create a `LibrepoSource{}` without functions or helpers.  It's not the preferred way to do it, but I think `pkg/osbuild/` should support that workflow generally.

Reply:
I can change them back to exported (there were in my first version), I changed them because it easy to non-working osbuild.LibrepoSources when doing this by hand, the repos and package IDs must match etc so this is meant as a hint "use-the-helper". But fine to change them as the rest of the package exports all types (I guess consistency here is more important than my whims)

Reply:
I changed it back to export everything. I would love to reduce our API surface though longer-term but not a discussion we need for this PR :)